src/sharedComponents/Header.js

```diff
@@ -18,16 +24,34 @@ class Header extends Component {
             See All Activities
           </NavItem>
         </Nav>
-        <Nav className="pull-right">
+
+        {this.determineButton()}
+
+      </Navbar>
+    );
+  }
+
+  determineButton = () => {
```

Comment:
determineButton could be named more informatively.  "loginLogoutButton", "userAuthButton" maybe?